x-pack/test/security_solution_api_integration/test_suites/detections_response/rules_management/prebuilt_rules/prebuilt_rule_customization/trial_license_complete_tier/configs/ess.config.ts

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { FtrConfigProviderContext } from '@kbn/test';
+
+export default async function ({ readConfigFile }: FtrConfigProviderContext) {
+  const functionalConfig = await readConfigFile(
+    require.resolve('../../../../../../../config/ess/config.base.trial')
+  );
+
+  const testConfig = {
+    ...functionalConfig.getAll(),
+    testFiles: [require.resolve('..')],
+    junit: {
+      reportName:
+        'Rules Management - Prebuilt Rule Customization Integration Tests - ESS Env - Trial License',
+    },
+  };
+  testConfig.kbnTestServer.serverArgs = testConfig.kbnTestServer.serverArgs.map((arg: string) => {
+    // Override the default value of `--xpack.securitySolution.enableExperimental` to enable the prebuilt rules customization feature
+    if (arg.includes('--xpack.securitySolution.enableExperimental')) {
+      return `--xpack.securitySolution.enableExperimental=${JSON.stringify([
+        'prebuiltRulesCustomizationEnabled',
+      ])}`;
+    }
+    return arg;
+  });
```

Comment:
This worked for me:
```ts
  const testConfig = {
    ...functionalConfig.getAll(),
    testFiles: [require.resolve('..')],
    junit: {
      reportName:
        'Rules Management - Prebuilt Rule Customization Integration Tests - ESS Env - Trial License',
    },
    kbnTestServer: {
      ...functionalConfig.get('kbnTestServer'),
      serverArgs: [
        ...functionalConfig.get('kbnTestServer.serverArgs'),
        `--xpack.securitySolution.enableExperimental=${JSON.stringify([
          'prebuiltRulesCustomizationEnabled',
        ])}`,
      ],
    },
  };
```
I would remove the additional check to simplify this, unless you still see it as necessary for some other reason.

Reply:
With this implementation, a new `--xpack.securitySolution.enableExperimental` flag is added to the config. I'm not entirely sure how this will be interpreted by the Kibana server since the same flag from the base config will also be present. The server will be started with something like this:

```
node path/to/kibana --xpack.securitySolution.enableExperimental=valuesFromBase --xpack.securitySolution.enableExperimental=prebuiltRulesCustomizationEnabled
```

Therefore, I'd prefer to have the flag entirely replaced for more transparency.